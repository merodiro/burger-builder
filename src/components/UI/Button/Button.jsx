import React from 'react'
import PropTypes from 'prop-types'

import classes from './Button.module.css'

const Button = ({ clicked, type, children }) => (
  <button
    onClick={clicked}
    className={[classes.Button, classes[type]].join(' ')}
  >
    {children}
  </button>
)

Button.prototype = {
  clicked: PropTypes.func,
  type: PropTypes.oneOf(['Success', 'Danger']).isRequired,
}

export default Button
