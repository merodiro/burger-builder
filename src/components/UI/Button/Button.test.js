import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import Button from './Button'

describe('<Button>', () => {
  it('renders the same output', () => {
    const component = shallow(<Button type="Success">Click me</Button>)

    expect(component).toMatchSnapshot()
  })

  it('renders the Danger type', () => {
    const component = shallow(<Button type="Danger">Cancel</Button>)

    expect(component).toMatchSnapshot()
  })
})
