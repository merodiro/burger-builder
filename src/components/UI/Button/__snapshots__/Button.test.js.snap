// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders the Danger type 1`] = `
<button
  className="Button Danger"
>
  Cancel
</button>
`;

exports[`<Button> renders the same output 1`] = `
<button
  className="Button Success"
>
  Click me
</button>
`;
